File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if(strs == null || strs.length == 0 ){
            return new ArrayList<>();
        }
        HashMap<String,List<String>> map = new HashMap<>();
        for( String s:strs){
            char[] NewArray = s.toCharArray();
            Arrays.sort(NewArray);
            String original = new String(NewArray);

            if(!map.containsKey(original)){
                map.put(original,new ArrayList<>());
            }
            map.get(original).add(s);
        
        }
        return new ArrayList<>(map.values());
    }
}

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group_anagram = {}

        for count in strs:

            char_count = [0] * 26

            for ch in count:
                char_count[ord(ch)-ord('a')]+=1
            
            char_count_tuple=tuple(char_count)

            if char_count_tuple in group_anagram:
                group_anagram[char_count_tuple].append(count)
            else:
                group_anagram[char_count_tuple]=[count]

        return list(group_anagram.values())